modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/util/test/CalendarUtilTest.java

```diff
@@ -204,6 +207,49 @@ null, createThemeDisplay(), calendarBookingInstance,
 		assertRepeatsForever(recurrence);
 	}
 
+	@Test
+	public void testToCalendarBookingJSONObjectVulnerabilities()
+		throws Exception {
+
+		ServiceContext serviceContext = createServiceContext();
+
+		HashMap<Locale, String> injectedContent = HashMapBuilder.put(
+			LocaleUtil.getDefault(),
+			"lp'\"></option><img src=x onerror=alert(document.location)>"
+		).build();
+
+		CalendarResource calendarResource =
+			_calendarResourceLocalService.addCalendarResource(
+				TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+				_classNameLocalService.getClassNameId(CalendarResource.class),
+				0, null, null, injectedContent,
+				RandomTestUtil.randomLocaleStringMap(), true, serviceContext);
```

Comment:
Hey @rijema please define the variable `injectedContent` inline since we're only escaping the calendarResource name.

Thanks!

Reply:
Done. Thanks!